app/controllers/transactions_controller.rb

```diff
@@ -1,63 +1,64 @@
+# frozen_string_literal: true
+
 class TransactionsController < ApplicationController
-  before_action :set_transaction, only: [:edit, :update, :destroy, :show]
+  before_action :authenticate_user!
+  before_action :set_transaction, only: %i[edit update destroy show]
 
   def index
-    @transactions = current_user.transactions.order(created_at: :desc).page(params[:page]).per(5)
+    @transactions = FetchTransactionsService.new(current_user, params).call
+    @accounts = current_user.accounts
   end
 
   def edit
     @accounts = current_user.accounts
+    @categories = Category.all
```

Comment:
Move this to `before_action` filter. It is being used in edit, show update and create.